Search box

meruyert93 opened the pull request:
I put my functions outside .then()

I would be really grateful to get your further advice, @mrTim

meruyert93 committed:
Further work on Store Locations

meruyert93 committed:
Functions Outside .then()

meruyert93 committed:
Cleaning code and Search-box Activation

meruyert93 commented:
Dear @mrTim, 

**Please, consider following changes:**

- More or less cleaned my code, deleted unnecessary comments etc,

- Separated scripts.js for each page

- Tried to make my search functionable, but it doesnot work, more details I described in Slack :)

**p.s. Vanya, I know you would not be 100% happy with my code structure, and I feel there are still some unneccessary empty lines to remove, unnecessary comments but I promise I will make it further perfect:)** 
**p.p.s Nevetherless, I hope it looks more better :D**

MrTim commented on .vscode/launch.json:
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
.vscode folder should be listed in gitignore file and should not be deployed. Because it's your own configuration for only your VScode.

MrTim commented on js/scripts-1.js:
```diff
@@ -0,0 +1,133 @@
+//Animation text
```
scripts-1.js? Really? :) 
mention that this a super bad name for a file is like saying nothing...
File naming is super important and you need to create a habit of naming them in meaningful way (same as branches in git for instance).

In this case you can name file: `landing.js` and `supermarkets.js` for ex. You can make other names, but they need to dive an idea what is inside.

MrTim commented on js/scripts-1.js:
```diff
@@ -0,0 +1,133 @@
+      return htmlRepresentation;
+  })
+  .then(htmlRepresetation =>{
+            contentDiv.innerHTML = htmlRepresetation;
```
too much of indentation. Please move a bit left.

MrTim commented on js/scripts-2.js:
```diff
@@ -0,0 +1,230 @@
+
+/**Connecting Map Location API  */
+    /*Creating google map*/
+    function initMap() {
```
this should be aligned left.

MrTim commented on js/scripts-2.js:
```diff
@@ -0,0 +1,230 @@
+/*Connecting Supermarkets locations API from EDEKA */
+const LocationApiUrl = 'https://www.edeka.de/api/marketsearch/markets?searchstring=Munchen&size=124';
+fetch(LocationApiUrl)
+.then(response => response.json())
```
fetch could looks like that:
```js
fetch()
  .then()
  .then()
  .catch()
```

MrTim commented on js/scripts-2.js:
```diff
@@ -0,0 +1,230 @@
+    setOnClickListener();
+    searchByPlace();
+    // searchstores();
+});     
```
what will happen if an error will happen in during api call or manipulation with data?

MrTim commented on js/scripts-2.js:
```diff
@@ -0,0 +1,230 @@
+function showStoresMarker(data) {
+    let bounds = new google.maps.LatLngBounds();
+    for (let i = 0; i < data.markets.length; i++){
+        let Lat = data.markets[i].coordinates.lat;
```
variable and function names should follow camelCase pattern

MrTim commented on js/scripts-2.js:
```diff
@@ -0,0 +1,230 @@
+//     }
+// };
+
+    function searchByPlace()  {
```
this needs to be aligned left

MrTim commented on js/scripts-2.js:
```diff
@@ -0,0 +1,230 @@
+            }
+          });
+
+          showStoresMarker(data);
```
where the `data` is coming from here? I think this will throw an error in the console.

MrTim commented on js/scripts-2.js:
```diff
@@ -0,0 +1,230 @@
+
+     processButtonSearch();
+
+    function processButtonSearch() {
```
code below should be aligned left

MrTim reviewed (changes requested):
Now it's better for sure! :+1: 
But please adjust code in regards comments I left and then we can move forward and talk about open questions :)

meruyert93 committed:
Searching markets by BTN

meruyert93 commented:
Dear **@MrTim,**

**Please, consider following changes:**
-Adding new functions such as searchMarketsNear() and triggerBtn()
-Considered all previous comments
-added catch(error) 
-further minor changes;

Thanks a lot in advance:)

MrTim commented on js/supermarkets.js:
```diff
@@ -3,30 +3,56 @@ let infoWindow;
+function triggerBtn() {
+    input = document.getElementById('inputValue');
+    input.addEventListener("keyup", function(event) {
+        if (event.keyCode === 13) {
```
keyCode is deprecated - https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
you need to use key instead - https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

MrTim commented on js/supermarkets.js:
```diff
@@ -3,30 +3,56 @@ let infoWindow;
+    input.addEventListener("keyup", function(event) {
+        if (event.keyCode === 13) {
+            event.preventDefault();
+            document.getElementById("myBtn").click();
```
this is not the way how this should be done.
Idea when you react on Enter is not to fire click from code, but to execute the function which button is using upon click.

MrTim commented on js/supermarkets.js:
```diff
@@ -3,30 +3,56 @@ let infoWindow;
+    input = document.getElementById('inputValue');
+    input.addEventListener("keyup", function(event) {
+        if (event.keyCode === 13) {
+            event.preventDefault();
```
you don;t need to prevent default here.

MrTim commented on js/supermarkets.js:
```diff
@@ -3,30 +3,56 @@ let infoWindow;
+        searchedPlace = 'München, Deutschland';
+    }
+    
+    const LocationApiUrl = `https://www.edeka.de/api/marketsearch/markets?searchstring=${searchedPlace}&size=124`;
```
you need to name variables in camelCase way.

MrTim commented on js/supermarkets.js:
```diff
@@ -3,30 +3,56 @@ let infoWindow;
+
+/*Searching markets by ZIP code or region */
+function searchMarketsNear(searchedPlace) {
+    input = document.getElementById('inputValue').value;
```
you're creating variable without saying is it let or const.

MrTim commented on js/supermarkets.js:
```diff
@@ -3,30 +3,56 @@ let infoWindow;
+/*Searching markets by ZIP code or region */
+function searchMarketsNear(searchedPlace) {
+    input = document.getElementById('inputValue').value;
+    if (input !== null && input !== '') {
```
could input value be null in any case?

MrTim commented on js/supermarkets.js:
```diff
@@ -148,76 +174,21 @@ function setOnClickListener() {
+function autoCompleteInSearch()  {
+    let input = document.getElementById('inputValue');
+    let autocomplete = new google.maps.places.Autocomplete(input);
+        autocomplete.bindTo('bounds', map);
```
indentation

MrTim commented on js/supermarkets.js:
```diff
@@ -148,76 +174,21 @@ function setOnClickListener() {
-        markers.length = 0;
+function clearLocations() {
+    infoWindow.close();
+    for (var i = 0; i < markers.length; i++) {
```
again you're using var.
If you don't want to remember those you can use `forEach` array method.

MrTim commented on js/supermarkets.js:
```diff
@@ -148,76 +174,21 @@ function setOnClickListener() {
+function clearLocations() {
+    infoWindow.close();
+    for (var i = 0; i < markers.length; i++) {
+    markers[i].setMap(null);
```
indentation.

MrTim commented on js/supermarkets.js:
```diff
@@ -148,76 +174,21 @@ function setOnClickListener() {
+    for (var i = 0; i < markers.length; i++) {
+    markers[i].setMap(null);
     }
+    markers.length = 0;
```
if you want to clear markers array you need to say `markers = [];`

meruyert93 committed:
Trigger Function and NullHide

meruyert93 commented:
**Dear @MrTim,

Please, consider following changes:**
-Considered all your previous comments
-added new function triggerFunction() in supermarkets.js
-Restructured code in index.js and made separate functions and called them inside .then()
-added new function hideNullElement() in index.js but it is no functionable:(, Can you help, please, with that? 
Thanks a lot in advance:)

MrTim commented on .vscode/launch.json:
```diff
@@ -5,11 +5,12 @@
     "configurations": [
         {
-            "type": "chrome",
+            "name": "Launch Chrome",
```
Again - this file should not be in included in git commits. You need to list it in `.gitignore` file - https://git-scm.com/docs/gitignore

MrTim commented on debug.log:
```diff
@@ -0,0 +1 @@
+[1119/193527.818:ERROR:settings.cc(320)] Settings magic is not 1129342067
```
all log files should be listed in gitignore as well

MrTim commented on js/index.js:
```diff
@@ -59,12 +59,19 @@ var TxtType = function(el, toRotate, period) {
-      content.docs.forEach(element => {
+    .then(response => response.json())
+    .then(data => {
+            displayProducts(data)
```
indentation

MrTim commented on js/index.js:
```diff
@@ -76,7 +83,7 @@ fetch(apiUrl)
                     <ul class="list-group list-group-flush">
                     <li class="list-group-item font-weight-bold">Price: <span class="text-danger h4">${element.preis} Euro</span></spand></li>
-                    <li class="list-group-item text-muted">Basic Price: <span class="text-danger">${element.basicPrice}</span></spand></li>
+                    <li class="list-group-item list-price text-muted" id="basic-price">Basic Price: <span class="text-danger">${element.basicPrice}</span></spand></li>
```
here you will have the same exact id for each and every element in the data.docs but `id` attribute should be always unique

MrTim commented on js/supermarkets.js:
```diff
@@ -47,7 +46,7 @@ function searchMarketsNear(searchedPlace) {
             setOnClickListener();
             autoCompleteInSearch()
-            triggerBtn();
+            triggerFunction();
```
what is expected behavior when this line gets executed? 
As far as I can see only addEventListener will be added for the input to be able to react on click, but this could be done without extra function.

MrTim commented on js/supermarkets.js:
```diff
@@ -18,27 +19,25 @@ function initMap() {
+    if (inputForSearch !== '') {
+        searchedPlace = inputForSearch;
     } else {
         searchedPlace = 'München, Deutschland';
```
this line will always be executed when your input is empty. So the parameter which you can pass inside this function will never be used as a value for search supermarkets.

meruyert93 committed:
Hiding Null and Triggering Enter

meruyert93 commented:
**Dear @MrTim,

Please, consider following changes:**

-Handling "null" with ternary operator;

-Considered all your comments above;

-Executing Enter key without function

-Adding EventListener to my Button;

-Some minor styling changes;

MrTim commented on css/styles.css:
```diff
@@ -183,4 +193,13 @@ body {
 }
 
+/*Hidden List with Null  */
+.yesList {
```
you don't need both classes and names are not really self explained. 
I can propose to use something like: 
```js
.hidden {
  visibility: hidden;
}
```

MrTim commented on js/index.js:
```diff
@@ -74,30 +74,28 @@ function displayProducts(data) {
                     <ul class="list-group list-group-flush">
                     <li class="list-group-item font-weight-bold">Price: <span class="text-danger h4">${element.preis} Euro</span></spand></li>
-                    <li class="list-group-item list-price text-muted" id="basic-price">Basic Price: <span class="text-danger">${element.basicPrice}</span></spand></li>
+                    <li class="list-group-item text-muted ${element.basicPrice ? "yesList" : "noList" }">Basic Price: <span class="text-danger">${element.basicPrice}</span></spand></li>
```
here you can apply only one class to only hide element and do nothing if it's fine to be displayed.

MrTim commented on js/supermarkets.js:
```diff
@@ -17,25 +17,24 @@ function initMap() {
-        if (e.key === 'Enter') {
+/*Triggering Function when Enter is pressed */
+input.addEventListener("keyup", function(e) {
+        if (e.key === 'Enter' && input.value !== '') {
```
something wrong with indentation here

MrTim commented on js/supermarkets.js:
```diff
@@ -17,25 +17,24 @@ function initMap() {
+});
+
+//Adding eventListener to my Button
+function clickmyBtn () {
```
two things:
1. how it's different from adding event listener for input?
2. this is defenetely not a proper name for the function. Becasuse what it does is to add listener and not click a button.

MrTim commented on js/supermarkets.js:
```diff
@@ -186,7 +182,7 @@ function clearLocations() {
 }
 
-/**ATTENTION! CODE belo is just temporary stored, it will be deleted */
+/**ATTENTION! CODE below is just temporary stored, it will be deleted */
```
When this code will be deleted? If you plan to merge this pr into master then you need to delete all commented code stored here temporarily.

meruyert93 committed:
Input Event and Design

meruyert93 commented:
**Dear @MrTim,

Please consider following changes:**
-Considering all your previously mentioned comments
-Making more design changes in both pages

MrTim commented on index.html:
```diff
@@ -59,7 +59,7 @@ <h1>
     </header>
 
-    <div class="container">
+    <div class="container" style="background-size: 100%; max-width: 100%; background-image:url('images/bag.jpg');">
```
try to not use inline styling. You can create a `.container` css class and put all of those there.

MrTim commented on index.html:
```diff
@@ -74,18 +74,19 @@ <h1>Find all discounts </h1>
     
-    <div class="container mt-5 mb-5" id="products" style="max-width: 95%; display: block;">
+    
+    <div class="container mt-5 mb-5" id="products" style="max-width: 95%; display:">
```
I think that style should be broken. At least `display:` part.
And again: you can move it to some css class.

MrTim commented on supermarkets.html:
```diff
@@ -58,9 +58,9 @@ <h1>
     </header>
 
-    <div class="container-fluid mb-5">
+    <div class="container-fluid" style="background-color: #FFF8C1; height: 800px;">
```
same here about styling.

MrTim commented on supermarkets.html:
```diff
@@ -58,9 +58,9 @@ <h1>
+    <div class="container-fluid" style="background-color: #FFF8C1; height: 800px;">
         <div class="row ">
-            <div class="col-4 mr-5 ml-5" style="background-color: #ddd; height: 750px; width:100%" >
+            <div class="col-4 mr-5 ml-5" style="background-color: transparent; height: 750px; width:100%" >
```
and here

MrTim commented on supermarkets.html:
```diff
@@ -97,13 +97,13 @@ <h1>
             </div>
-            <div class="col-7" style="background-color: darkgrey; height: 750px; width:100%" >
-                <div id="map" class="w-100" style="height: 750px;">
+            <div class="col-7 mt-3" style="background-color: transparent; height: 750px; width:100%" >
```
here and line below as well.

MrTim commented on js/supermarkets.js:
```diff
@@ -17,21 +17,19 @@ function initMap() {
-function clickmyBtn () {
-    document.getElementById("myBtn").addEventListener("click", function() {
+//Adding eventListener to Input
+    input.addEventListener('input', function() {
```
I think somethign is wrong with indentation here.

MrTim commented on js/supermarkets.js:
```diff
@@ -180,65 +178,5 @@ function clearLocations() {
             markers[i].setMap(null);
     }
             markers = [];
```
indentation problem here, please move it left. And line below as well.

MrTim reviewed (changes requested):
Good! please fix ONLY those points which I mentioned in my review and send it again. We want to close it and all future work should happen in new branch!

meruyert93 committed:
Cleaning code

meruyert93 commented:
**Dear @MrTim,**

**Considered all your comments above and fixed only those comments.**

**Another question:** **_once you approve this PR, how to resolve conflict before merging?_**

Thanks a lot in advance!

MrTim reviewed (approved):
Ok, now all looks good! So you can merged it now. About your question:
please look at this article about the conflicts: https://www.atlassian.com/git/tutorials/using-branches/merge-conflicts.

How you can approach this merge:

1. switch to a master branch locally - `git checkout master`
2. do `git fetch && git pull origin master`
3. try to merge: `git merge --no-ff origin/search-box`
4. look at the output and see which files has conflicts
5. go to those files and do what is explained in the article
6. make sure that your application is working
7. then add all files and commit and push them.
8. Profit!
